tests/unit/package_managers/test_pip.py

```diff
@@ -105,23 +112,22 @@ def test_get_pip_metadata(
     expect_name = toml_name or py_name or cfg_name
     expect_version = toml_version or py_version or cfg_version
 
-    if expect_name and expect_version:
+    if expect_name:
         name, version = pip._get_pip_metadata(PKG_DIR)
 
         assert name == expect_name
         assert version == expect_version
     else:
-        with pytest.raises(PackageRejected) as exc_info:
-            pip._get_pip_metadata(PKG_DIR)
-
-        if expect_name:
-            missing = "version"
-        elif expect_version:
-            missing = "name"
+        if repo_exists:
+            mock_get_repo_id.return_value = MOCK_REPO_ID
+            name, _ = pip._get_pip_metadata(PKG_DIR_SUBPATH)
+            assert name == "bar-subpath"
+            assert "Processing metadata from git repository" in caplog.text
         else:
-            missing = "name, version"
+            with pytest.raises(PackageRejected) as exc_info:
+                pip._get_pip_metadata(PKG_DIR)
+            assert str(exc_info.value) == "Could not take name from the repository origin url"
```

Comment:
@slimreaper35 if you can't figure out why the function is not raising an exception here, check what it returns

```suggestion
            retval = pip._get_pip_metadata(PKG_DIR)
            print(retval)
            assert False
```

Reply:
You might also want to add some `print`s to the get_pip_metadata function to see where the name is coming

Or use pdb or some other debugger, I don't really know how to use debuggers TBH :smile: